app/api/server/lib/email-channel.js

```diff
@@ -0,0 +1,63 @@
+import { EmailChannel as EmailChannelRaw } from '../../../models/server/raw';
+import { EmailChannel, Users } from '../../../models';
+import { hasPermissionAsync } from '../../../authorization/server/functions/hasPermission';
+
+export async function find({ userId, query = {}, pagination: { offset, count, sort } }) {
+	if (!await hasPermissionAsync(userId, 'manage-email-channels')) {
+		throw new Error('error-not-allowed');
+	}
+	const cursor = EmailChannelRaw.find(query, {
+		sort: sort || { name: 1 },
+		skip: offset,
+		limit: count,
+	});
+
+	const total = await cursor.count();
+
+	const emailChannels = await cursor.toArray();
+
+	return {
+		emailChannels,
+		count: emailChannels.length,
+		offset,
+		total,
+	};
+}
+
+export async function findOne({ userId, id }) {
+	if (!hasPermissionAsync(userId, 'manage-email-channels')) {
+		throw new Error('error-not-allowed');
+	}
+	return EmailChannel.findOneById(id);
+}
+
+export async function createEdit(emailChannelParams, userId) {
+	if (!hasPermissionAsync(userId, 'manage-email-channels')) {
+		throw new Error('error-not-allowed');
+	}
+
+	const updateEmailChannel = {
```

Comment:
Why are you declaring this const here if it might not be used when a new record is created?